File: VB/CustomReportModel.vb
Imports DevExpress.XtraReports.Wizards
Imports System.Drawing.Printing

Namespace CustomReportWizard
    Friend Class CustomReportModel
        Inherits XtraReportModel

        Public Property PaperKind() As PaperKind
        Public Property PageMargins() As Margins

        Public Sub New()
            PaperKind = PaperKind.Letter
            PageMargins = New Margins(100, 100, 100, 100)
        End Sub
        Public Sub New(ByVal model As CustomReportModel)
            MyBase.New(model)
            PaperKind = model.PaperKind
            PageMargins = New Margins(model.PageMargins.Left, model.PageMargins.Right, model.PageMargins.Top, model.PageMargins.Bottom)
        End Sub

        Public Overrides Function Clone() As Object
            Return New CustomReportModel(Me)
        End Function

        Public Overrides Function Equals(ByVal obj As Object) As Boolean
            Dim other = TryCast(obj, CustomReportModel)
            Return other IsNot Nothing AndAlso MyBase.Equals(obj) AndAlso PaperKind = other.PaperKind AndAlso PageMargins Is other.PageMargins
        End Function

        Public Overrides Function GetHashCode() As Integer
            Return MyBase.GetHashCode() Xor PaperKind.GetHashCode() Xor PageMargins.GetHashCode()
        End Function
    End Class
End Namespace

File: CS/CustomReportModel.cs
using DevExpress.XtraReports.Wizards;
using System.Drawing.Printing;

namespace CustomReportWizard {
    class CustomReportModel : XtraReportModel {
        public PaperKind PaperKind { get; set; }
        public Margins PageMargins { get; set; }

        public CustomReportModel() {
            PaperKind = PaperKind.Letter;
            PageMargins = new Margins(100, 100, 100, 100);
        }
        public CustomReportModel(CustomReportModel model) : base(model) {
            PaperKind = model.PaperKind;
            PageMargins = new Margins(model.PageMargins.Left, model.PageMargins.Right, model.PageMargins.Top, model.PageMargins.Bottom);
        }

        public override object Clone() {
            return new CustomReportModel(this);
        }

        public override bool Equals(object obj) {
            var other = obj as CustomReportModel;
            return other != null
                && base.Equals(obj)
                && PaperKind == other.PaperKind
                && PageMargins == other.PageMargins;
        }

        public override int GetHashCode() {
            return base.GetHashCode() ^ PaperKind.GetHashCode() ^ PageMargins.GetHashCode();
        }
    }
}
